File: MCA-I-MiniProject/fptp-backend-api/src/main/java/com/fptp/api/models/ParkingVehicle.java
package com.fptp.api.models;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.*;

import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ParkingVehicle {
    private int ParkingId;
    private String Name;
    private String MobileNo;

    @Pattern(
            regexp = "^[A-Za-z]{2}[0-9]{1,2}[A-Za-z]{0,3}[0-9]{4}$",
            message = "Invalid vehicle number format. Example: TN01AB1234"
    )
    private String VehicleNo;

    private LocalDateTime CheckIn;
    private LocalDateTime CheckOut;
    private String TotalHours;
    private int VehicleTypeId;
    private double PerHourCharge;
    private String TransactionId;
    private String PaidThrow;
    private Boolean IsPaid;
    private LocalDateTime CreatedDate;

    private LocalDateTime ModifiedDate;
    @NotBlank(message = "Owner Id is required")
    private String OwnerId;
    private boolean IsDeleted;


    private double TotalCharge;
    @NotBlank(message = "Block name is required")
    private String BlockName;

    private  int SlotNo;

    private Boolean IsSubmittedKey;
    private String VehicleTypeName;
    private String Comments;
}

File: 1_Semester/MiniProject/first-park-then-pay/src/app/core/models/parkingvehicle.ts
import { FormControl } from "@angular/forms";

export interface Parkingvehicle {
    parkingId:number,
    name:string,
    mobileNo:string,
    vehicleNo:string,
    checkIn:Date,
    checkOut:Date,
    totalHours:string,
    vehicleTypeId:number,
    perHourCharge:number,
    transactionId:string,
    paidThrow:string,
    isPaid:boolean,
    createdDate:Date,
    modifiedDate:Date,
    ownerId:string,
    isDeleted:boolean,
    totalCharge:number,
    blockName:string,
    slotNo:number,
    isSubmittedKey:boolean,
    vehicleTypeName:string,
    comments:string
}

export interface ParkingvehicleForm {
  parkingId: FormControl<number | null>;
  name: FormControl<string | null>;
  mobileNo: FormControl<string | null>;
  vehicleNo: FormControl<string | null>;
  checkIn: FormControl<Date | null>;
  checkOut: FormControl<Date | null>;
  totalHours: FormControl<string | null>;
  vehicleTypeId: FormControl<number | null>;
  perHourCharge: FormControl<number | null>;
  transactionId: FormControl<string | null>;
  paidThrow: FormControl<string | null>;
  isPaid: FormControl<boolean | null>;
  createdDate: FormControl<Date | null>;
  modifiedDate: FormControl<Date | null>;
  ownerId: FormControl<string | null>;
  isDeleted: FormControl<boolean | null>;
  totalCharge: FormControl<number | null>;
  blockName: FormControl<string | null>;
  slotNo: FormControl<number | null>;
  isSubmittedKey: FormControl<boolean | null>;
  vehicleTypeName: FormControl<string | null>;
  comments: FormControl<string | null>;
}
